### PR TITLE
Set up build configurations and references for GSA10.1 build

### PR DESCRIPTION
**Pull request by IsakNaslundBh**
### Issues addressed by this PR

Handles initial setup of #242 

 

- Adding gsa dll reference for GSA 10.1 to libs folder
- Setting up build configs for version 8.7 and 10.1
- Updating usings where needed
- setting up different adapter name dependant on build config

 ### Test files

 ### Changelog

 ### Additional comments

Should build as before for the previous Debug and Release builds, with GSAAdapter being the resulting class name. Will look into removing this and version GSAAdapter to the 8.7 build after 4.2 beta has been released.

@FraserGreenroyd adding you in for general compliance check. DO not expect you to test the code, just making sure I am not adding anything here that will be in the way for the release. The intention is that only the new builds should be different from before, while the default builds should remain unaffected

### CODE REVIEW

**Review by annastigenberg — DISMISSED** [review]

**Comment by annastigenberg** [comment]
Tested various old scripts using the GSA adapter, using push/ pull/ update etc, after compiling the 'debug' build. All work well.

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]
The suggested change can be ignored if you want.

My actual request for changes would be to include an `altConfigs.txt` file (see an example [here](https://github.com/BHoM/Revit_Toolkit/blob/master/altConfigs.txt)) that would allow BHoMBot to check both configurations in a `check core` check (this doesn't impact the installers).

Can I confirm the default build at the moment is still targeting 8_7 (does impact installers) (which is also the purpose of the first code comment).

Otherwise, from a CI perspective I'm happy with this.

File: GSA_Adapter/AdapterActions/Execute.cs [hunk 1]
```diff
@@ -28,7 +28,13 @@
 
 namespace BH.Adapter.GSA
 {
+#if GSA_10_1
+    public partial class GSA_10_1Adapter
+#elif GSA_8_7
+    public partial class GSA_8_7Adapter
+#else
```
[the remark this answers is not in the document]
Comment (IsakNaslundBh) [reply GSA_Adapter/AdapterActions/Execute.cs:35]: The default build is targeting GSA 8.7. WHat I did though is that I did _not_ define any constants for the default debug and release builds, so they should still revert to the `#else` case. This essentially means that we will have two identical builds. GSAAdapter and GSA8_7Adapter will be doing the exact same thing.

My reasoning for setting it up this way is that I did not want this PR to impact the beta release in any way at all. Plan (which I will document on the issue if we agree with this strategy) is that we remove the GSAAdapter in milestone 4.3, with proper versioning and everything, and only have the versioned builds left. Basically, remove default Debug and Release builds and remove the else statement.

Does that make sense?

[the remark this answers is not in the document]
Comment (FraserGreenroyd) [reply GSA_Adapter/AdapterActions/Execute.cs:35]: That does make sense.

If we don't change the configs for the installer (which is what we're recommending here, only adding one for `core`) then the 4.2 beta will be unchanged, so then doing a full upgrade in 4.3 with versioning is probably a good plan, I'd support that as a roll out.

**Comment by FraserGreenroyd** [comment]
@BHoMBot check required

**Review by FraserGreenroyd — APPROVED** [review]
Approved with discussion with @IsakNaslundBh

**Pull request merged**